Framework/Algorithms/src/SortXAxis.cpp

```diff
@@ -51,15 +51,25 @@ void SortXAxis::init() {
       boost::make_shared<StringListValidator>(orderingValues);
   declareProperty("Ordering", orderingValues[0], orderingValidator,
                   "Ascending or descending sorting", Direction::Input);
+  declareProperty("IgnoreHistogramValidation", false,
+                  "This will stop SortXAxis from throwing if the workspace is "
+                  "not a valid histogram for this algorithm to work on.");
```

Comment:
Does not the option `Ordering` help you in this case?

- the method *determineIfHistogramIsValid* seems to be wrong to me: histograms are sorted already but can be sorted reversely and point data can be unsorted -> in my opinion this can be deleted?

Explanation of valid/invalid histogram missing.

`SortXAxis` should be able generally to handle these cases and a new input option seems to be redundant.